packages/amplify-cli/src/domain/telemetry.ts

```diff
@@ -0,0 +1,85 @@
+import uuid from 'uuid';
+import os from 'os';
+import { Input } from './input';
+import https from 'https';
+import url, { UrlWithStringQuery } from 'url';
+import redactInput from './identifiable-input-regex';
+
+export class Telemetry {
+  SessionUuid: String;
+  InstallationUuid: String;
+  Version: String;
+  Url: UrlWithStringQuery;
+
+  constructor(installationUuid: String, version: String) {
+    this.InstallationUuid = installationUuid;
+    this.Version = version;
+    this.SessionUuid = uuid.v4();
+    this.Url = url.parse('https://aws-amplify-cli-telemetry.us-east-1.amazonaws.com/metrics');
+  }
+
+  async emit(input: Input, error: Error | null, state: String, waitForSend: boolean = false) {
+    const payload = new TelemetryPayload(this.SessionUuid, this.InstallationUuid, this.Version, input, error, state);
+    waitForSend ? await this.send(payload) : this.send(payload);
+  }
+
+  async send(payload: TelemetryPayload) {
+    return new Promise((resolve, _) => {
+      const data = JSON.stringify(payload);
+      const req = https.request({
+        hostname: this.Url.hostname,
+        port: this.Url.port,
+        path: this.Url.path,
+        method: 'POST',
+        headers: {
+          'content-type': 'application/json',
+          'content-length': data.length,
+        },
+      });
+      req.on('error', () => {});
+      req.write(data);
+      req.end(() => {
+        resolve();
+      });
+    });
+  }
+}
+
+export const WorkFlowState = {
+  Successfull: 'SUCCEEDED',
```

Comment:
typo: `Successful`, but if we need the literal values on the backend, then call the enum members the same.